### PR TITLE
New App: Live on Twitch

### PR DESCRIPTION
**Pull request by daltonclaybrook**
# Description
This PR adds a new app called "Live on Twitch." After configuring the app with your Twitch account via OAuth2, the app shows a list of the streamers you follow who are currently live. If only one streamer is live, or if a streamer has gone live in the last two minutes, that streamer is shown in a special "recently live" widget. If none of your followed streamers are live, the app does not appear in the rotation.

## When multiple followed streamers are live

live_list

## When only one followed streamer is live, or when a streamer has just gone live minutes ago

recently_live

# Copilot

### Summary
🎮📺🟣

This pull request adds a new applet called Live on Twitch, which shows the user's followed streamers who are live on Twitch. The pull request includes the applet code in `live_on_twitch.star` and the applet manifest in `manifest.yaml`. The applet code uses various modules and features to interact with the Twitch API and display the streamer information on the Tidbyt device.

> _`Live on Twitch` applet_
> _Shows streamers in the fall_
> _Code uses `Twitch API`_

### Walkthrough
* Create a new applet called Live on Twitch that shows the user's followed streamers who are live on Twitch ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283), [link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-021633cd08b710cf60220c28f06dffa37eece64f6708c3c5c9a5dd14984dfdaaR1-R8))
  * Use the Twitch API and OAuth2 module to authenticate the user and fetch the live streamers data ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283))
  * Use the caching module to store and update the data periodically ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283))
  * Use the rendering module to display the streamers' names, titles, and logos on the Tidbyt device ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283))
  * Use the schema module to define the applet settings and inputs, such as the refresh interval and the OAuth2 token ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283))
  * Use constants and comments to document and configure the applet behavior and parameters ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283))
  * Add the `live_on_twitch.star` file that contains the applet code ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-7230e91031664febc4723c0557b54bf7f537a0383896bb62dd62051a6a5531caR1-R283))
  * Add the `manifest.yaml` file that contains the applet metadata for the Tidbyt app store ([link](https://github.com/tidbyt/community/pull/1894/files?diff=unified&w=0#diff-021633cd08b710cf60220c28f06dffa37eece64f6708c3c5c9a5dd14984dfdaaR1-R8))

### CODE REVIEW

**Comment by daltonclaybrook** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by noahpodgurski** [comment]
Can we get this merged please @matslina

**Comment by matslina** [comment]
Hey, thanks for the PR.
We're requesting new oauth2 credentials from Twitch for this app. It apparently has to go through some review process so sit tight while we're waiting for that.

**Comment by daltonclaybrook** [comment]
Gotcha, thanks for the update @matslina.

**Comment by daltonclaybrook** [comment]
@matslina Any update on this? It has been over a month. Pretty discouraging as a developer who is new to this community and was hoping to make several contributions.

File: apps/liveontwitch/live_on_twitch.star [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+load("time.star", "time")
+
+TWITCH_CLIENT_ID = "h980fr8jmfwp9gkcdo0sc1m251qlw9"
```
Comment (matslina) [inline apps/liveontwitch/live_on_twitch.star:16]:
```suggestion
TWITCH_CLIENT_ID = secret.decrypt("AV6+xWcEw1WPTIFrqcknBcgMQkpBeoeS/Hq4jUi6YarnMTNolrRnRXMfK0Nb4V32klRP36EOjrX5AFohuSx4WKqwOGKPBkqf93bn43H/+ymL7y+yiJJd1wwvaV4uGEdF/nKZeqH8560uDy0YBoXKHb6z0m9jR7GTCktyJtG3xua9VBp9")
```

File: apps/liveontwitch/live_on_twitch.star [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+TWITCH_CLIENT_SECRET = secret.decrypt("AV6+xWcEQV17TbSvb4u35Ir6ac/CuqEVYoo+XRIDH/FftFQhjn+VSh3oqchdz1hAbco3PsD5BAqyoL7hQHpT7iS0l5KIpYfcP84rBJOhp6or2SS38OfqjglBPcKQHSV9kAZEcgU7wIVBpJzO7I9R1d57YxDLhEpOB1O60OYLuc/8gsEI")
```
Comment (matslina) [inline apps/liveontwitch/live_on_twitch.star:17]:
```suggestion
TWITCH_CLIENT_SECRET = secret.decrypt("AV6+xWcEaGyyQ31L8I3zSauUNqXvLAsuS4H/WZuX8qSFM0pGA8/l6xqPnAhvzHOMvKpmwus3e3bLBFxP2evfLsCA1CJKZ+52BCAyvhPTyZBBRfDAlp+J16Ebk5JOLezMlX7pcZ3OyLgB8UB+WuMkZBLuFVHbPzQTYRjdYl/GYNoJCXBi")
```

File: apps/liveontwitch/live_on_twitch.star (hunk 1 above)
Comment (matslina) [inline apps/liveontwitch/live_on_twitch.star:16]:
```suggestion
TWITCH_CLIENT_ID = secret.decrypt("AV6+xWcE/aYwEfmThAFhVm8NJrh/c9tjxuvENfNesy1lUWIqQo9GGq90ENm/UDJwJ+1zpnJvESzP/GxVAt7+YoHiUJfJ2mCleNpdFsMJ0ix58jEEvdWH6H6MJ8dI0nSOKl5uCsXSkXpGGvh5UgI8pxlMfvKHg59D5FOeX62GB/2PqSaJ")
```

**Comment by matslina** [comment]
Sorry for the delay. We've been swamped, and especially OAuth2 apps tend to take longer for us to get through the pipe.

File: apps/liveontwitch/live_on_twitch.star [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+load("time.star", "time")
+
+TWITCH_CLIENT_ID = secret.decrypt("AV6+xWcE/aYwEfmThAFhVm8NJrh/c9tjxuvENfNesy1lUWIqQo9GGq90ENm/UDJwJ+1zpnJvESzP/GxVAt7+YoHiUJfJ2mCleNpdFsMJ0ix58jEEvdWH6H6MJ8dI0nSOKl5uCsXSkXpGGvh5UgI8pxlMfvKHg59D5FOeX62GB/2PqSaJ")
```
Comment (matslina) [inline apps/liveontwitch/live_on_twitch.star:16]:
```suggestion
TWITCH_CLIENT_ID = secret.decrypt("t7n82ips6t16w5hmf27abea5ypqa2a")
```

File: apps/liveontwitch/live_on_twitch.star [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+load("time.star", "time")
+
+TWITCH_CLIENT_ID = secret.decrypt("t7n82ips6t16w5hmf27abea5ypqa2a")
```
Comment (matslina) [inline apps/liveontwitch/live_on_twitch.star:16]:
```suggestion
TWITCH_CLIENT_ID = "t7n82ips6t16w5hmf27abea5ypqa2a"
```
ffs

**Comment by matslina** [comment]
Hopefully this works. Shoot me a DM on Discord if not and I'll take a closer look on Monday. Promise!

**Comment by daltonclaybrook** [comment]
Awesome! Thanks so much @matslina!

**Pull request merged**